File: leetcode/algorithms/459_repeated_substring_pattern/main.java
class RepeatedSubstringPattern {
    public boolean repeatedSubstringPattern(String s) {
        return s.substring(1).concat(s.substring(0, s.length() - 1)).contains(s);
    }


    // Best Solution
    public boolean bestSolution(String s) {
        int l = s.length();
        for (int i = l / 2; i >= 1; i--) {
            if (l % i == 0) {
                int m = l / i;
                String subS = s.substring(0, i);
                StringBuilder sb = new StringBuilder();
                for (int j = 0; j < m; j++) {
                    sb.append(subS);
                }
                if (sb.toString().equals(s))
                    return true;
            }
        }
        return false;
    }
}

File: leetcode/algorithms/459_repeated_substring_pattern/main.dart
class RepeatedSubstringPattern {
  bool repeatedSubstringPattern(String s) {
    return (s + s).substring(1, s.length * 2 - 1).contains(s);
  }


  // Solution
  bool solution(String s) {
    return (s + s).substring(1, s.length * 2 - 1).contains(s);
  }
}
